packages/utils/telemetry-utils/src/errorLogging.ts

```diff
@@ -275,22 +305,17 @@ function getValidTelemetryProps(obj: any, keysToOmit: Set<string>): ITelemetryPr
             continue;
         }
         const val = obj[key];
-        switch (typeof val) {
-            case "string":
-            case "number":
-            case "boolean":
-            case "undefined":
-                props[key] = val;
-                break;
-            default: {
-                if (isTaggedTelemetryPropertyValue(val)) {
-                    props[key] = val;
-                } else {
-                    // We don't support logging arbitrary objects
-                    props[key] = "REDACTED (arbitrary object)";
-                }
-                break;
-            }
+        // ensure only valid props get logged, since props of logging error could be in any shape
+        const validProp = filterValidTelemetryProps(val);
+        const validTaggedProp = filterValidTelemetryProps(val.value);
```

Comment:
Don't you need to do `val?.value` in case it's null or undefined?  Might be good to get Unit Test coverage.

Reply:
yes i actually just changed it to checking if it's taggedProp first, then filter. But Vlad has another way above, moving redacted directly into filter function. I will follow that